tempesta_fw/sock_srv.c

```diff
@@ -596,6 +596,26 @@ static int tfw_cfg_cns_retries, tfw_cfg_out_cns_retries;
 static unsigned int tfw_cfg_retry_nip, tfw_cfg_out_retry_nip;
 static unsigned int tfw_cfg_sticky_sess, tfw_cfg_out_sticky_sess;
 static unsigned int tfw_cfg_sg_flags, tfw_cfg_out_sg_flags;
+static struct {
+	bool sched		: 1;
+	bool queue_size		: 1;
+	bool fwd_timeout	: 1;
+	bool fwd_retries	: 1;
+	bool cns_retries	: 1;
+	bool retry_nip		: 1;
+	bool sticky_sess	: 1;
+} __attribute__((packed)) tfw_cfg_is_set;
+
+/* Please keep the condition in these two macros in sync. */
+#define TFW_CFGOP_HAS_DFLT(ce, v)				\
+	(tfw_cfg_is_dflt_value(ce) && tfw_cfg_is_set.v)
+#define TFW_CFGOP_INHERIT(ce, v)				\
+({								\
+	if (tfw_cfg_is_dflt_value(ce) && tfw_cfg_is_set.v) {	\
+		tfw_cfg_##v = tfw_cfg_out_##v;			\
```

Comment:
Seems, that line duplicates initialisation of parameters `tfw_cfg_##v` in `tfw_cfgop_begin_srv_group()` and not needed.

As a point of future improvement: if default value is passed to given cfg entry (`tfw_cfg_is_dflt_value(ce) == true`), but global default value is not set yet (`tfw_cfg_is_set.v == false`), we can parse that value and save both to global default server group option (`tfw_cfg_out_##v`) and server group option for the current group (`tfw_cfg_##v`). That will save time for the cases when a lot of server group with default settings are defined in config file.